### ISSUE TITLE
"Export" tab does not work with `composer update --no-dev`

### ISSUE THREAD

**Issue by danielmarschall** [phpmyadmin/phpmyadmin#17716]
We use the following to keep phpMyAdmin up-to-date:

```sh
# exit on errors
set -e

DIR=$(pwd)
INSTALL_DIR="phpmyadmin"

# First install
if [ ! -d "$DIR/$INSTALL_DIR" ]; then
        cd "$DIR"
        git clone --branch=STABLE https://github.com/phpmyadmin/phpmyadmin.git "$INSTALL_DIR"
fi

# Installation/Update
cd "$DIR"/"$INSTALL_DIR"

# switch to STABLE
git checkout STABLE
# package.json gets modified. We need to undo that, otherwise "git pull" won't work
git reset --hard
# remove unversioned files
git clean -fd
# fetch HEAD
git pull --quiet

# Install phpMyAdmin
composer update
yarn install --production
scripts/generate-mo --quiet

# Foreign language recompile
# Requires: sudo aptitude install gettext
# Attention: the cwd MUST be phpmyadmin!
scripts/generate-mo --quiet

# Copy custom phpMyAdmin configuration if it exists
if [ -d "$DIR"/config.inc.php ]; then
        cp "$DIR"/config.inc.php "$DIR"/"$INSTALL_DIR"/
fi

# Now remove the thousands of yarn tmp folders that are never deleted!
rm -rf /tmp/yarn--*
```

based on https://docs.phpmyadmin.net/en/latest/setup.html#installing-from-git

It turns out if we change `composer update` to `composer update --no-dev` , then the "Export" tab does not work anymore, probably because a dependency is missing.

**Comment by williamdes** [comment]
Hi @danielmarschall
I am working on this currently
Your report looks the same as  #17034

**Comment by danielmarschall** [comment]
Hello @williamdes 
Thank you for your quick reply.
It is probably related. I am not sure if it is the same.
For me, clicking the "Export" tab does nothing - no error message, just a very short "wait" cursor and then nothing happens. The JavaScript console shows no error, and the Network tab does not show any HTTP 500 error (all requests are HTTP 200), and also the PHP error log at the server does not show anything. This is very strange.

**Comment by williamdes** [comment]
Hello @danielmarschall 

> It turns out if we change composer update to composer update --no-dev , then the "Export" tab does not work anymore, probably because a dependency is missing.

Absolutely

you should do 
```sh
composer install --optimize-autoloader
yarn install --production --frozen-lockfile
```

And `git pull --quiet --ff`

I see no reason to make an `update` since we provide for users a `composer.lock` on this branch, right ?

**Comment by danielmarschall** [comment]
@williamdes Hello,
just a small update: I have applied the changes for git, composer and yarn you mentioned ( [my updated script](https://misc.daniel-marschall.de/code/bash/update_pma.sh) ) and there are no errors.
`composer install --optimize-autoloader` works, but `composer install --optimize-autoloader --no-dev` does not yet.

**Comment by williamdes** [comment]
I think I have figured out that the error you are reporting is https://github.com/phpmyadmin/phpmyadmin/issues/17519 and if you change the branch in your script to `QA_5_2` and use `--no-dev` it will work. It works on work workstation with the changes implemented. Let me know

**Comment by danielmarschall** [comment]
@williamdes Thank you very much. I can confirm that it works now!
I have also fixed my script accordingly. Is there an approximate time when this change is put into the STABLE branch? Will it happen with the release of >=5.2.1 or earlier?

**Comment by williamdes** [comment]
> @williamdes Thank you very much. I can confirm that it works now! I have also fixed my script accordingly. Is there an approximate time when this change is put into the STABLE branch? Will it happen with the release of >=5.2.1 or earlier?

Awesome!

About time I have no idea, but for sure it will happen as soon as 5.2.1 is out: the STABLE branch will be updated

I am closing this issue as a duplicate
Thank you for patience and good reporting

**Comment by williamdes** [comment]
Duplicate of #17519
